### PR TITLE
wapm-cli: init at 0.5.0

### PR DESCRIPTION
**Pull request by lucperkins**
###### Motivation for this change

Adds a package for [wapm](https://docs.wasmer.io/ecosystem/wapm), the WebAssembly package manager.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/package-management/wapm/cli/default.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+, openssl
+, rustPlatform
+, Security
+, stdenv
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "wapm-cli";
+  version = "0.5.0";
+  cargoSha256 = "1bx9rv6bkmkpysz0zfcwfv88r47d5nbzw0zwakqs3n7mdsyb4q2x";
```
Comment (SuperSandro2000) [inline pkgs/tools/package-management/wapm/cli/default.nix:11]:
```suggestion
```

File: pkgs/tools/package-management/wapm/cli/default.nix [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+  src = fetchFromGitHub {
+    owner = "wasmerio";
+    repo = "wapm-cli";
+    rev = "v${version}";
+    sha256 = "0w0ck60xm47b93y803i2b2bpv7nsahzvzda3ll7pjg88f8qz5fx1";
+  };
+
```
Comment (SuperSandro2000) [inline pkgs/tools/package-management/wapm/cli/default.nix:19]:
```suggestion

  cargoSha256 = "1bx9rv6bkmkpysz0zfcwfv88r47d5nbzw0zwakqs3n7mdsyb4q2x";

```

File: pkgs/tools/package-management/wapm/cli/default.nix [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  buildInputs = []
+    ++ stdenv.lib.optionals stdenv.isLinux [ openssl ]
+    ++ stdenv.lib.optionals stdenv.isDarwin [ Security ];
```
Comment (SuperSandro2000) [inline pkgs/tools/package-management/wapm/cli/default.nix:22]:
```suggestion
  buildInputs = stdenv.lib.optionals stdenv.isLinux [ openssl ]
    ++ stdenv.lib.optionals stdenv.isDarwin [ Security ];
```

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 94675` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>wapm-cli</li>
  </ul>
</details>

```
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking sources
unpacking source archive /nix/store/wa0y04cwic9xjbm7ifdv44lswnip5jcg-source
source root is source
unpacking source archive /nix/store/f23987ql247qq0n59b580sq2a6k93kik-wapm-cli-0.5.0-vendor.tar.gz
@nix { "action": "setPhase", "phase": "patchPhase" }
patching sources
Validating consistency between /build/source/Cargo.lock and /build/wapm-cli-0.5.0-vendor.tar.gz/Cargo.lock
@nix { "action": "setPhase", "phase": "configurePhase" }
configuring
@nix { "action": "setPhase", "phase": "buildPhase" }
building
++ env CC_x86_64-unknown-linux-gnu=/nix/store/b96dqbx6pri2xp2xxlq6i269virrdaw6-gcc-wrapper-9.3.0/bin/cc CXX_x86_64-unknown-linux-gnu=/nix/store/b96dqbx6pri2xp2xxlq6i269virrdaw6-gcc-wrapper-9.3.0/bin/c++ CC_x86_64-unknown-linux-gnu=/nix/store/b96dqbx6pri2xp2xxlq6i269virrdaw6-gcc-wrapper-9.3.0/bin/cc CXX_x86_64-unknown-linux-gnu=/nix/store/b96dqbx6pri2xp2xxlq6i269virrdaw6-gcc-wrapper-9.3.0/bin/c++ cargo build -j 2 --release --target x86_64-unknown-linux-gnu --frozen
[0m[0m[1m[32m   Compiling[0m libc v0.2.67
[0m[0m[1m[32m   Compiling[0m proc-macro2 v1.0.9
[0m[0m[1m[32m   Compiling[0m unicode-xid v0.2.0
[0m[0m[1m[32m   Compiling[0m syn v1.0.16
[0m[0m[1m[32m   Compiling[0m cfg-if v0.1.10
[0m[0m[1m[32m   Compiling[0m memchr v2.3.3
[0m[0m[1m[32m   Compiling[0m cc v1.0.50
[0m[0m[1m[32m   Compiling[0m serde v1.0.104
[0m[0m[1m[32m   Compiling[0m autocfg v1.0.0
[0m[0m[1m[32m   Compiling[0m version_check v0.9.1
[0m[0m[1m[32m   Compiling[0m lazy_static v1.4.0
[0m[0m[1m[32m   Compiling[0m byteorder v1.3.4
[0m[0m[1m[32m   Compiling[0m ryu v1.0.2
[0m[0m[1m[32m   Compiling[0m pkg-config v0.3.17
[0m[0m[1m[32m   Compiling[0m log v0.4.8
[0m[0m[1m[32m   Compiling[0m itoa v0.4.5
[0m[0m[1m[32m   Compiling[0m typenum v1.11.2
[0m[0m[1m[32m   Compiling[0m bytes v0.5.4
[0m[0m[1m[32m   Compiling[0m bitflags v1.2.1
[0m[0m[1m[32m   Compiling[0m fnv v1.0.6
[0m[0m[1m[32m   Compiling[0m slab v0.4.2
[0m[0m[1m[32m   Compiling[0m failure_derive v0.1.7
[0m[0m[1m[32m   Compiling[0m pin-project-lite v0.1.4
[0m[0m[1m[32m   Compiling[0m futures-core v0.3.4
[0m[0m[1m[32m   Compiling[0m getrandom v0.1.14
[0m[0m[1m[32m   Compiling[0m proc-macro2 v0.4.30
[0m[0m[1m[32m   Compiling[0m semver-parser v0.7.0
[0m[0m[1m[32m   Compiling[0m rustc-demangle v0.1.16
[0m[0m[1m[32m   Compiling[0m foreign-types-shared v0.1.1
[0m[0m[1m[32m   Compiling[0m unicode-xid v0.1.0
[0m[0m[1m[32m   Compiling[0m openssl v0.10.28
[0m[0m[1m[32m   Compiling[0m void v1.0.2
[0m[0m[1m[32m   Compiling[0m unicode-width v0.1.7
[0m[0m[1m[32m   Compiling[0m unicode-segmentation v1.6.0
[0m[0m[1m[32m   Compiling[0m ascii v0.9.3
[0m[0m[1m[32m   Compiling[0m syn v0.15.44
[0m[0m[1m[32m   Compiling[0m httparse v1.3.4
[0m[0m[1m[32m   Compiling[0m subtle v1.0.0
[0m[0m[1m[32m   Compiling[0m crc32fast v1.2.0
[0m[0m[1m[32m   Compiling[0m matches v0.1.8
[0m[0m[1m[32m   Compiling[0m either v1.5.3
[0m[0m[1m[32m   Compiling[0m arrayvec v0.4.12
[0m[0m[1m[32m   Compiling[0m futures-sink v0.3.4
[0m[0m[1m[32m   Compiling[0m futures-task v0.3.4
[0m[0m[1m[32m   Compiling[0m futures-io v0.3.4
[0m[0m[1m[32m   Compiling[0m smallvec v1.2.0
[0m[0m[1m[32m   Compiling[0m byte-tools v0.3.1
[0m[0m[1m[32m   Compiling[0m pin-utils v0.1.0-alpha.4
[0m[0m[1m[32m   Compiling[0m native-tls v0.2.4
[0m[0m[1m[32m   Compiling[0m adler32 v1.0.4
[0m[0m[1m[32m   Compiling[0m try-lock v0.2.2
[0m[0m[1m[32m   Compiling[0m ppv-lite86 v0.2.6
[0m[0m[1m[32m   Compiling[0m openssl-probe v0.1.2
[0m[0m[1m[32m   Compiling[0m nodrop v0.1.14
[0m[0m[1m[32m   Compiling[0m encoding_rs v0.8.22
[0m[0m[1m[32m   Compiling[0m regex-syntax v0.6.16
[0m[0m[1m[32m   Compiling[0m static_assertions v0.3.4
[0m[0m[1m[32m   Compiling[0m tower-service v0.3.0
[0m[0m[1m[32m   Compiling[0m doc-comment v0.3.2
[0m[0m[1m[32m   Compiling[0m opaque-debug v0.2.3
[0m[0m[1m[32m   Compiling[0m percent-encoding v2.1.0
[0m[0m[1m[32m   Compiling[0m fake-simd v0.1.2
[0m[0m[1m[32m   Compiling[0m vec_map v0.8.1
[0m[0m[1m[32m   Compiling[0m dtoa v0.4.5
[0m[0m[1m[32m   Compiling[0m base64 v0.11.0
[0m[0m[1m[32m   Compiling[0m remove_dir_all v0.5.2
[0m[0m[1m[32m   Compiling[0m strsim v0.8.0
[0m[0m[1m[32m   Compiling[0m mime v0.3.16
[0m[0m[1m[32m   Compiling[0m ansi_term v0.11.0
[0m[0m[1m[32m   Compiling[0m linked-hash-map v0.5.2
[0m[0m[1m[32m   Compiling[0m fallible-streaming-iterator v0.1.9
[0m[0m[1m[32m   Compiling[0m encode_unicode v0.3.6
[0m[0m[1m[32m   Compiling[0m wasmparser v0.51.4
[0m[0m[1m[32m   Compiling[0m fallible-iterator v0.2.0
[0m[0m[1m[32m   Compiling[0m whoami v0.5.3
[0m[0m[1m[32m   Compiling[0m license-exprs v1.5.0
[0m[0m[1m[32m   Compiling[0m openssl-src v111.6.1+1.1.1d
[0m[0m[1m[32m   Compiling[0m indexmap v1.3.2
[0m[0m[1m[32m   Compiling[0m num-traits v0.2.11
[0m[0m[1m[32m   Compiling[0m num-integer v0.1.42
[0m[0m[1m[32m   Compiling[0m unicase v2.6.0
[0m[0m[1m[32m   Compiling[0m proc-macro-error-attr v0.4.11
[0m[0m[1m[32m   Compiling[0m proc-macro-error v0.4.11
[0m[0m[1m[32m   Compiling[0m nom v5.1.1
[0m[0m[1m[32m   Compiling[0m backtrace-sys v0.1.34
[0m[0m[1m[32m   Compiling[0m thread_local v1.0.1
[0m[0m[1m[32m   Compiling[0m libsqlite3-sys v0.17.1
[0m[0m[1m[32m   Compiling[0m http v0.2.0
[0m[0m[1m[32m   Compiling[0m futures-channel v0.3.4
[0m[0m[1m[32m   Compiling[0m foreign-types v0.3.2
[0m[0m[1m[32m   Compiling[0m unreachable v1.0.0
[0m[0m[1m[32m   Compiling[0m textwrap v0.11.0
[0m[0m[1m[32m   Compiling[0m heck v0.3.1
[0m[0m[1m[32m   Compiling[0m unicode-bidi v0.3.4
[0m[0m[1m[32m   Compiling[0m unicode-normalization v0.1.12
[0m[0m[1m[32m   Compiling[0m block-padding v0.1.5
[0m[0m[1m[32m   Compiling[0m miniz_oxide v0.3.6
[0m[0m[1m[32m   Compiling[0m lru-cache v0.1.2
[0m[0m[1m[32m   Compiling[0m openssl-sys v0.9.54
[0m[0m[1m[32m   Compiling[0m http-body v0.3.1
[0m[0m[1m[32m   Compiling[0m idna v0.2.0
[0m[0m[1m[32m   Compiling[0m iovec v0.1.4
[0m[0m[1m[32m   Compiling[0m net2 v0.2.33
[0m[0m[1m[32m   Compiling[0m num_cpus v1.12.0
[0m[0m[1m[32m   Compiling[0m time v0.1.42
[0m[0m[1m[32m   Compiling[0m atty v0.2.14
[0m[0m[1m[32m   Compiling[0m termios v0.3.1
[0m[0m[1m[32m   Compiling[0m clicolors-control v1.0.1
[0m[0m[1m[32m   Compiling[0m dirs v1.0.5
[0m[0m[1m[32m   Compiling[0m rpassword v4.0.5
[0m[0m[1m[32m   Compiling[0m filetime v0.2.8
[0m[0m[1m[32m   Compiling[0m xattr v0.2.2
[0m[0m[1m[32m   Compiling[0m quote v1.0.3
[0m[0m[1m[32m   Compiling[0m futures-util v0.3.4
[0m[0m[1m[32m   Compiling[0m aho-corasick v0.7.10
[0m[0m[1m[32m   Compiling[0m csv-core v0.1.10
[0m[0m[1m[31merror[0m[1m:[0m failed to run custom build command for `openssl-sys v0.9.54`

Caused by:
  process didn't exit successfully: `/build/source/target/release/build/openssl-sys-cd92ead8a923101a/build-script-main` (exit code: 101)
  --- stdout
  cargo:rustc-cfg=const_fn
  CC_x86_64-unknown-linux-gnu = Some("/nix/store/b96dqbx6pri2xp2xxlq6i269virrdaw6-gcc-wrapper-9.3.0/bin/cc")
  CFLAGS_x86_64-unknown-linux-gnu = None
  CFLAGS_x86_64_unknown_linux_gnu = None
  HOST_CFLAGS = None
  CFLAGS = None
  CRATE_CC_NO_DEFAULTS = None
  DEBUG = Some("false")
  CARGO_CFG_TARGET_FEATURE = Some("fxsr,sse,sse2")
  running "perl" "./Configure" "--prefix=/build/source/target/x86_64-unknown-linux-gnu/release/build/openssl-sys-b58785276e9412d3/out/openssl-build/install" "no-dso" "no-ssl3" "no-unit-test" "no-comp" "no-zlib" "no-zlib-dynamic" "no-engine" "no-async" "no-shared" "linux-x86_64" "-O2" "-ffunction-sections" "-fdata-sections" "-fPIC" "-m64"

  --- stderr
  thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Os { code: 2, kind: NotFound, message: "No such file or directory" }', /build/wapm-cli-0.5.0-vendor.tar.gz/openssl-src/src/lib.rs:363:39
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
[0m[0m[1m[33mwarning[0m[1m:[0m build failed, waiting for other jobs to finish...
[0m[0m[1m[31merror[0m[1m:[0m build failed
```

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 94675` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>wapm-cli</li>
  </ul>
</details>

File: pkgs/tools/package-management/wapm/cli/default.nix [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+  buildInputs = stdenv.lib.optionals stdenv.isLinux [ openssl ]
+    ++ stdenv.lib.optionals stdenv.isDarwin [ Security ];
+
+  meta = with stdenv.lib; {
+    description = "A package manager for WebAssembly modules";
+    homepage = "https://docs.wasmer.io/ecosystem/wapm";
+    license = with licenses; [ mit ];
+    maintainers = [ maintainers.lucperkins ];
+    platforms = platforms.all;
```
Comment (SuperSandro2000) [inline pkgs/tools/package-management/wapm/cli/default.nix:27]:
```suggestion
```

File: pkgs/tools/package-management/wapm/cli/default.nix [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+  src = fetchFromGitHub {
+    owner = "wasmerio";
+    repo = "wapm-cli";
+    rev = "v${version}";
+    sha256 = "0w0ck60xm47b93y803i2b2bpv7nsahzvzda3ll7pjg88f8qz5fx1";
+  };
+
+  buildInputs = stdenv.lib.optionals stdenv.isLinux [ openssl ]
+    ++ stdenv.lib.optionals stdenv.isDarwin [ Security ];
```
Comment (SuperSandro2000) [inline pkgs/tools/package-management/wapm/cli/default.nix:20]:
```suggestion
  buildInputs = lib.optionals stdenv.isLinux [ openssl ]
    ++ lib.optionals stdenv.isDarwin [ Security ];
```

File: pkgs/tools/package-management/wapm/cli/default.nix [hunk 6]
```diff
@@ -0,0 +1,29 @@
+{ fetchFromGitHub
+, openssl
+, rustPlatform
+, Security
+, stdenv
+}:
```
Comment (SuperSandro2000) [inline pkgs/tools/package-management/wapm/cli/default.nix:6]:
```suggestion
{ stdenv
, lib
, fetchFromGitHub
, openssl
, rustPlatform
, Security
}:
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
@ofborg eval

**Comment by ghishadow** [comment]
hey thanks for packaging wapm-cli , is anything blocking this merge ? is there anything i can do to help in merging this

**Comment by SuperSandro2000** [comment]
The last review comments I left are blocking this. Feel free to take the commits, apply the changes and open a new PR.

**Comment by lucperkins** [comment]
I'm going to follow up with a fresh PR in a few minutes here.

**Pull request closed without merging**